### PR TITLE
Added missing options method and headers

### PR DESCRIPTION
**Pull request by rzanluchi**
Still breaking a lot of tests :(

### CODE REVIEW

File: src/app.cr [hunk 1]
```diff
@@ -29,7 +30,8 @@ before_all "/todos" do |env|
   # Support CORS and set responses to JSON as default.
   headers env, {
     "Access-Control-Allow-Origin" => "*",
-    "Content-Type" => "application/json"
+    "Content-Type" => "application/json",
+    "access-control-allow-headers" => "Content-Type"
```
Comment (mauricioabreu) [inline src/app.cr]: I think * is better for the problem no? Also the Kemal source code uses this headers capitalized no?

File: shard.yml [hunk 2]
```diff
@@ -1,4 +1,4 @@
-name: todo
+name: app
```
Comment (mauricioabreu) [inline shard.yml]: I think `todo` is a better name for the app no?

File: src/app.cr [hunk 3]
```diff
@@ -21,6 +21,7 @@ def repr(todo, base_url)
   {"uid"   => todo._id,
     "title" => todo.title,
     "order" => todo.order,
+    "completed" => todo.completed,
```
Comment (mauricioabreu) [inline src/app.cr]: :

File: src/app.cr (hunk 1 above)
Comment (mauricioabreu) [inline src/app.cr]: https://github.com/sdogruyol/kemal/blob/56067b628097193d9491dc72a3b05204a19e7e74/spec/helpers_spec.cr#L76

Comment (rzanluchi) [inline src/app.cr]: didn't get it, you pointed to another header, these are allowed headers. But yeah * could be better I will changed it

File: shard.yml (hunk 2 above)
Comment (rzanluchi) [inline shard.yml]: for heroku this name should be the same as the src/app.cr to it knows how to compile and run the application, just for that I changed

**Comment by rzanluchi** [comment]
There is a lot of broken tests and I can't figure out why they are failing, the logs on heroku look wierd

File: shard.yml (hunk 2 above)
Comment (mauricioabreu) [inline shard.yml]: hm ok! `todo_app` can be an option though.

File: src/app.cr (hunk 1 above)
Comment (mauricioabreu) [inline src/app.cr]: Cool! Does the way it is written affects the headers being sent?

Comment (mauricioabreu) [inline src/app.cr]: It looks like Kemal uses "This-Way-Of-Writing-Headers".

**Comment by rzanluchi** [comment]
Now its ok

File: src/todo_app.cr [hunk 4]
```diff
@@ -29,20 +30,41 @@ before_all "/todos" do |env|
...
+  }
+end
 
 get "/" do |env|
   env.redirect TODOS_URI
 end
 
+options "/todos" do |env|
+  headers env, {
+    "access-control-allow-methods" => "GET,HEAD,POST,DELETE,OPTIONS,PUT,PATCH"
```
Comment (mauricioabreu) [inline src/todo_app.cr]: Capitalize it here and merge. \o/ Good job! :+1:

**Comment by mauricioabreu** [comment]
Just fix that little issue and we are ready to go! :+1:

File: src/todo_app.cr (hunk 4 above)
Comment (rzanluchi) [inline src/todo_app.cr]: forgot about this one :(

**Pull request merged**